### DISCUSSION TITLE
addHarmony function works for subset data but not the whole data

### DISCUSSION THREAD

**Discussion by TauHHCC** [GreenleafLab/ArchR#1765] [Questions / Documentation]
Hi
I have a question about correct batch effect using addHarmony function.
I have multiple time points and several biological replicates for each time points. When I do batch correction for them together, it didn't work, but if I subset the data to each time points, the corrections with addHarmony works well. However, since it works after dimension reduce, then you cannot simply combine each corrected time points together again right?
How do you solve this kind of issue?

Best

**Comment by rcorces** [comment]
Sorry "it didnt work" is not enough detail for me to comment. Did it give an error? Or did it not give the results you wanted? If the latter, I would recommend reading the Harmony documentation and then the ArchR function parameters for `addHarmony()` to see how you can tweak your call to `addHarmony()`. Note that `...` pass-through arguments which allow you to specify additional parameters that will get directly passed to the `harmony::HarmonyMatrix()` function. This is how you would customize the behavior of harmony through ArchR to best suit your own project.

**Comment by TauHHCC** [comment]
Hi, I mean whatever if I run addHarmony(), it always give same UMAP plot. The biological replicates at same time points stay separately. 
Here in this UMAP, cluster 3 and 4 are two biological batches both for time point 0. addHarmony() didn't put them together, but when I isolate data for cluster 3 and 4 and then cluster them again, they can mix well (updata: I just tested again, it looks when data set is simple, it can mix well by itself even though without addHarmony() (addHarmony() didn't make difference) ). I have tried majority of the arguments, it so far has not made difference. 
![Screenshot_2022-12-06_15-41-34](https://user-images.githubusercontent.com/113095697/205942231-3aac0eb3-2ea6-4cf1-9915-a94158819353.png)

**Reply by rcorces** [reply]
I'm afraid I dont have too much to add. ArchR just provides a wrapper around harmony. If reading the harmony documentation does not help you to solve the behavior you are seeing, then there isnt much more that ArchR can do.
